controller/workitem.go

```diff
@@ -87,7 +85,7 @@ func (c *WorkitemController) authorizeWorkitemTypeEditor(ctx context.Context, sp
 	if space != nil && editorID == space.OwnerID.String() {
 		return true, nil
 	}
-	return false, errors.NewUnauthorizedError("user is not allowed to change workitem type")
+	return false, errors.NewForbiddenError("user is not a workitem creator or space owner")
```

Comment:
Why did you change this error type?

Reply:
This should be a ForbiddenError. The workitem could not be deleted because the user doesn't have necessary permissions.

Reply:
I know what it means but I really don't see a difference between: the action is forbidden and you're not authorized.

Reply:
@kwk @jarifibrahim let's discuss this in weekly today? I have some thoughts around it.

Reply:
Based on the discussion in the weekly sync, we decided to use `NewForbiddenError` specific for this case.